### PR TITLE
jmeter: 3.3 -> 4.0

### PR DESCRIPTION
**Pull request by markus1189**
[x] Tested execution of binary file

### CODE REVIEW

**Comment by volth** [comment]
I am not sure that ```$ARGS``` interpolated correctly in lines [13-14](https://github.com/markus1189/nixpkgs/blob/a48a376f36833817c5d99ffc37b9094e061deed1/pkgs/applications/networking/jmeter/default.nix#L13)

It looks like it is interpolated to empty string

**Comment by markus1189** [comment]
I think you're right @volth, I am currently investigating it seems like there is nothing to substitute anyway because the script does already in 3.3 no longer have this line.

**Comment by markus1189** [comment]
@volth Okay I refactored the package a bit, could you make another review pass?

Things done:

1) extracted `version` attribute
2) added test in checkPhase whether jmeter runs (I had to reorder the phases, true?)
3) prefixed some of the `bin/` scripts with the `jmeter-` prefix
4) wrapped the jmeter program because 4.0 checks JAVA_HOME anyway

File: pkgs/applications/networking/jmeter/default.nix [hunk 1]
```diff
@@ -1,19 +1,37 @@
...
   src = fetchurl {
     url = "http://archive.apache.org/dist/jmeter/binaries/apache-${name}.tgz";
-    sha256 = "190k6yrh5casadphkv4azp4nvf4wf2q85mrfysw67r9d96nb9kk5";
+    sha256 = "1dvngvi6j8qb6nmf5a3gpi5wxck4xisj41qkrj8sjwb1f8jq6nw4";
   };
 
-  buildInputs = [ jre ];
+  buildInputs = [ jre makeWrapper ];
+
+  phases = [ "unpackPhase" "installPhase" "checkPhase" "fixupPhase" ];
```
Comment (fpletz) [inline pkgs/applications/networking/jmeter/default.nix]: Please use `dontBuild = true` instead.

File: pkgs/applications/networking/jmeter/default.nix [hunk 2]
```diff
@@ -1,19 +1,37 @@
...
+  name = "jmeter-${version}";
+  version = "4.0";
   src = fetchurl {
     url = "http://archive.apache.org/dist/jmeter/binaries/apache-${name}.tgz";
-    sha256 = "190k6yrh5casadphkv4azp4nvf4wf2q85mrfysw67r9d96nb9kk5";
+    sha256 = "1dvngvi6j8qb6nmf5a3gpi5wxck4xisj41qkrj8sjwb1f8jq6nw4";
   };
 
-  buildInputs = [ jre ];
+  buildInputs = [ jre makeWrapper ];
```
Comment (fpletz) [inline pkgs/applications/networking/jmeter/default.nix]: `makeWrapper` should be in `nativeBuildInputs`.

File: pkgs/applications/networking/jmeter/default.nix (hunk 1 above)
Comment (markus1189) [reply pkgs/applications/networking/jmeter/default.nix]: I also had to change the order of the `checkPhase` though

**Comment by volth** [comment]
As I remember it should be safe to skip ```buildPhase``` without declaring its absence in ```phases``` or ```doNotBuild```

**Comment by markus1189** [comment]
@volhovm Do you refer to the `dontBuild = true` comment?

I can add that, but I still need that line because I am reordering the `checkPhase` to be before the `installPhase`.  Otherwise the check does not work.  Should I instead perform the `wrapProgram` in some other phase?

**Comment by volth** [comment]
@markus1189 this: https://github.com/markus1189/nixpkgs/pull/2/files

**Comment by markus1189** [comment]
@volth Already included in the latest commit.  Thanks for the pr that is much ncier

**Comment by Mic92** [comment]
there is a bit more wrapping needed:
```console
$ /nix/store/34wdgmwq6gqhzbyzx18sig78jh4yrygz-jmeter-4.0/bin/jmeter-create-rmi-keystore.sh: line 18: keytool: command not found
Copy the generated rmi_keystore.jks to jmeter/bin folder or reference it in property 'server.rmi.ssl.keystore.file'
$ /nix/store/34wdgmwq6gqhzbyzx18sig78jh4yrygz-jmeter-4.0/bin/jmeter-heapdump.sh: line 24: java: command not found
$ /nix/store/34wdgmwq6gqhzbyzx18sig78jh4yrygz-jmeter-4.0/bin/jmeter-mirror-server.sh: line 27: java: command not found
$ /nix/store/34wdgmwq6gqhzbyzx18sig78jh4yrygz-jmeter-4.0/bin/jmeter-mirror-server.sh: line 27: java: command not found
```

**Comment by markus1189** [comment]
Done ;)

I really like how this came out, it's a good improvement of the old package, thanks for all the feedback!

**Comment by Mic92** [comment]
the following are still missing java:

```console
$ /nix/store/6867z1rhini3q3qwps2d8brd2k49r2xs-jmeter-4.0/bin/jmeter-shutdown.sh: line 24: java: command not found
$ /nix/store/6867z1rhini3q3qwps2d8brd2k49r2xs-jmeter-4.0/bin/jmeter-stoptest.sh: line 24: java: command not found
```

**Comment by markus1189** [comment]
Updated and added more tests.

Is the use of `timeout` okay for the test of the jmeter-mirror-server?

**Comment by Mic92** [comment]
The test still has `java` in PATH, if you run it outside, it still misses it.

**Comment by markus1189** [comment]
Should be fixed now

**Pull request merged**